src/Masa.Utils.Exceptions/Handlers/GlobalExceptionFilter.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Utils.Exceptions.Handlers;
+
+/// <summary>
+/// Mvc pipeline exception filter to catch global exception
+/// </summary>
+public class GlobalExceptionFilter : IExceptionFilter
+{
+    private readonly MasaExceptionHandlerOptions _options;
+    private readonly MasaExceptionLogRelationOptions _logRelationOptions;
+    private readonly ILogger<GlobalExceptionFilter>? _logger;
+
+    public GlobalExceptionFilter(IOptions<MasaExceptionHandlerOptions> options,
+        IOptions<MasaExceptionLogRelationOptions> logRelationOptions,
+        ILogger<GlobalExceptionFilter>? logger = null)
+    {
+        _options = options.Value;
+        _logRelationOptions = logRelationOptions.Value;
+        _logger = logger;
+    }
+
+    public void OnException(ExceptionContext context)
+    {
+        var masaExceptionContext = new MasaExceptionContext(context.Exception, context.HttpContext);
+        _options.ExceptionHandler?.Invoke(masaExceptionContext);
+
+        if (masaExceptionContext.HttpContext.Response.HasStarted)
+            return;
+
+        if (masaExceptionContext.ExceptionHandled)
+        {
+            context.ExceptionHandled = true;
+            context.Result = new DefaultExceptionResult(
+                masaExceptionContext.Message,
+                masaExceptionContext.StatusCode,
+                masaExceptionContext.ContentType);
+            return;
+        }
+
+        _logger?.WriteLog(masaExceptionContext.Exception, LogLevel.Error, _logRelationOptions);
```

Comment:
LogLevel

Reply:
I don't understand what you mean

Reply:
Log level should not be error.

Reply:
UserFriendlyException defaults to Information, other exceptions default to Error